main.tf

```
@@ -1,24 +1,36 @@
 /**
- * Creates and configures an S3 bucket for storing logs from various AWS
- * services and enables CloudTrail on all regions. Logs will expire after a
- * default of 90 days. Includes support for sending CloudTrail events to a
- * CloudWatch Logs group.
+ * Supports two main uses cases:
  *
- * Logging from the following services is supported:
+ * 1. Creates and configures a single private S3 bucket for storing logs from various AWS services, which are nested as bucket prefixes, and enables CloudTrail on all regions. Logs will expire after a default of 90 days, with option to configure retention value. Includes support for sending CloudTrail events to a CloudWatch Logs group.
+ *
+ * 2. Creates and configures a single private S3 bucket for a single AWS service, and enables CloudTrail on all regions. Logs will expire after a default of 90 days, with option to configure retention value. Includes support for sending CloudTrail events to a CloudWatch Logs group.
+ *
+ * Logging from the following services is supported for both cases:
  *
  * * [CloudTrail](https://aws.amazon.com/cloudtrail/)
  * * [Config](https://aws.amazon.com/config/)
- * * [Elastic Load Balancing](https://aws.amazon.com/elasticloadbalancing/)
+ * * [Elastic Load Balancing (and Application Load Balancing (ALB))](https://aws.amazon.com/elasticloadbalancing/)
```

AWS has made this very confusing because they use and _re-use_ these terms in different ways.

```suggestion
 * * [Classic Load Balancer (ELB) and Application Load Balancer (ALB)](https://aws.amazon.com/elasticloadbalancing/)
```